dj/api/nodes.py

```diff
@@ -214,7 +232,7 @@ def read_nodes(*, session: Session = Depends(get_session)) -> List[NodeMetadata]
     """
     List the available nodes.
     """
-    nodes = session.exec(select(Node).options(joinedload(Node.current))).all()
+    nodes = session.exec(select(Node).options(joinedload(Node.current))).unique().all()
```

Comment:
Hmm, why is `unique()` needed here? Isn't that already going to be the case given that we have a constraint on unique node names?

Reply:
Yeah I was surprised I need to do this as well. Let me know if there's a better solution but I was fighting this error once I added "cube_elements" to `NodeRevision`
```
sqlalchemy.exc.InvalidRequestError: The unique() method must be invoked on this Result, as it contains results that include joined eager loads against collections
```

Reply:
It looks like this change was added that requires `unique()`: https://github.com/sqlalchemy/sqlalchemy/issues/4395